src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -193,10 +193,16 @@ private UncommonFields CreateUncommonFields()
                 retVal._lazyObsoleteAttributeData = ObsoleteAttributeData.Uninitialized;
             }
 
-            if (!_packedFlags.IsUseSiteDiagnosticPopulated)
-            {
-                retVal._lazyUseSiteDiagnostic = CSDiagnosticInfo.EmptyErrorInfo; // Indicates unknown state.
-            }
+            //
+            // Do not set _lazyUseSiteDiagnostic !!!!
+            //
+            // "null" Indicates "no errors" or "unknown state".
+            // and we know which one of the states we have from IsUseSiteDiagnosticPopulated
+            //
+            // setting _lazyUseSiteDiagnostic to a sentinel value here would introduce
+            // a number of states with various permutations between IsUseSiteDiagnosticPopulated, UncommonFields and _lazyUseSiteDiagnostic
+            // some of them, in tight races, may lead to returning the sentinel as the diagnostics.
```

Comment:
In that change the comment will be more appropriate when the UncommonFields is created. I will add the comment there.

---
In reply to: [124596328](https://github.com/dotnet/roslyn/pull/20497#discussion_r124596328) [](ancestors = 124596328,124581343)